### ISSUE TITLE
[Feature request] Support for custom root paths

### ISSUE THREAD

**Issue by oskar-anderson** [ZarehD/AspNetStatic#11]
## Problem to solve
I am trying to use this library to generate static pages from a razor pages app to host on Github Pages. The problem is that the pages work locally, but not in GP, since it uses `www.{username}.github.io/{projectName}/` URL structure. This causes root paths to point to `www.{username}.github.io/`

I tried using:
```
var builder = WebApplication.CreateBuilder(new WebApplicationOptions()
{
    WebRootPath = "wwwroot/AwesomeProject"
});
```
and
```
app.UsePathBase("/projectName");
```
But it does not seem to do anything. I know Hugo allows to define the project base URL path in `config.toml`. Maybe I am missing some builtin functionality, but I don't seem to find a way to solve this currently.

## Proposal
Add support for replacing `~` with a custom BaseURL string for all paths beginning with `~/`.

**Comment by ZarehD** [comment]
The output destination folder has no bearing on the content of generated pages. It just specifies where the files for the static pages are to be placed.

The issue is with the ~ based relative paths; they don't work properly. Use "/my-page" instead of "~/my-page". This is not something that's unique to AspNetStatic, btw. Just use the AspNet tag helpers (_asp-*_ or _UrlHelper.*_) to generate the URL to your pages, and your links will work properly regardless of the site's base URL/domain.

**Comment by oskar-anderson** [comment]
I understand that the output directory must be `wwwroot`, since this would be the only way to develop the app in `dotnet watch` mode and then generate static pages that still work with all the static css, js files placed in `wwwroot/static`. Then everything other than `static` in `wwwroot` could be gitignored.

I tried using `asp-pages` tag, but it does not work. I created a demo of my workflow of what I am trying to achieve [here](https://github.com/oskar-anderson/AspNetStaticDemo). 

I would appreciate if you could have a look. Some instructions on running:
1. a) Clone the project or b) if the permission allow use my repo (I sent a collaborator invitation). 
1.a.1. Enable `Settings -> Actions -> General -> Workflow permissions -> Read and Write permissions`
1.a.2. Select gh-pages branch and root directory in `Settings -> Pages -> Build and deployment` 
1.a.3. Hopefully every push to master should now build it into gh-pages branch.
3. You can hard reset to 4ceffab5 since changes after that did not do anything

Running the project is done by `dotnet run --project WebApp` or `dotnet run --project WebApp -- stop` to only generate static pages. Generated files can be viewed by `python pyServer.py` in `wwwroot` directory. But most of the time if it works with `dotnet run` I would just push it to Github Pages and expect it to work.

**Comment by ZarehD** [comment]
Hi @oskar-anderson,

I cloned your AspNetStaticDemo repo, then setup an IIS website pointing to the wwwroot folder. I also took the content of wwwroot (sans pyServer.py) and copied them to another unrelated folder (c:\oskar), and set the IIS site to point to it. In both cases, it all works fine; all img/css/js content loads/displays as expected and all inter-page links work as expected. I don't see a problem here; at least not one that's related to AspNetStatic.

Also, the _asp-page_ tag helper in your Header.cshtml code...
```Razor
<a class="nav-link" asp-page="Contact">Contact</a>
```
...doesn't work b/c there's no Contact.cshtml page in your project!

**Comment by oskar-anderson** [comment]
This is a Github Pages thing, everything works fine locally.

I recreated the app in NextJS and had the same problem with defining a base URL for Github Pages. I tried out deployment on Vercel and everything works fine. I guess the moral is that Github Pages is 💩. 

The thing is that I came from a Hugo background and that allows to create shortcode markdown pre-processors. So markdown image link would look like this `![MyTV]({{< picturepath >}}DSC_1625.jpg)` that would point to `{{ .Page.Site.Params.picturepath }}.` that got its value from a constant config file. Links were relative to baseURL variable (with the value being either `https://oskar-anderson.github.io/RepoName/` or `localhost`) defined in the config file. This system works well with Github Pages.

The solution seems to run all markdown content through a templating engine that replaces {{ BASE_PATH }} with a value from appsettings.json before it gets served as a Razor Page. Also all links would have to use this BASE_PATH constant.

I realize now that my problem has nothing to do with this library. Sorry for wasting time.
